src/compiler/crystal/syntax/lexer.cr

```diff
@@ -1703,6 +1703,10 @@ module Crystal
               @reader.pos = old_pos
               @column_number = old_column
               next_string_token delimiter_state
+              value = @token.value
+              if value.is_a? String
```

Comment:
Is there a way for this not to be true? I think maybe we can do `@token.value = '\n' + @token.value.to_s`.